kvserver: fix assertion in qps-based rebalancing logic

aayushshah15 opened the pull request:
The qps-based rebalancing logic has an assertion at the end that ensures that
the qps delta between stores _after a rebalance_ is at least as good as the qps
delta between stores _before the rebalance_ (i.e. this assertion is supposed to
fire for any logic errors in our code).

This assertion was broken because we were computing the "current" delta (i.e.
before rebalance) incorrectly.

Release note: None

aayushshah15 committed:
kvserver: fix assertion in qps-based rebalancing logic

The qps-based rebalancing logic has an assertion at the end that ensures that
the qps delta between stores _after a rebalance_ is at least as good as the qps
delta between stores _before the rebalance_ (i.e. this assertion is supposed to
fire for any logic errors in our code).

This assertion was broken because we were computing the "current" delta (i.e.
before rebalance) incorrectly.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75773)
<!-- Reviewable:end -->

aayushshah15 committed:
kvserver: improve logging around successful qps-based rebalances

Previously, we weren't logging any debug-able details around successful
qps-based lease transfers and replica rebalances.

Release note: None

aayushshah15 commented:
TFTRs!

bors r+
